crates/sncast/src/state/hashing.rs

```diff
@@ -0,0 +1,84 @@
+use cairo_felt::Felt252;
+use itertools::Itertools;
+
+// if we change API this might have collisions with old API hashes
+#[allow(dead_code)]
+pub fn hash_script_subcommand_call(selector: &str, inputs: &[Felt252]) -> String {
+    let mut res = selector.as_bytes().to_owned();
+    let mut inputs_bytes: Vec<u8> = inputs.iter().flat_map(Felt252::to_bytes_be).collect_vec();
+    res.append(&mut inputs_bytes);
+    res.iter().map(|item| format!("{item:x?}")).join("").clone()
```

Comment:
Technically speaking - is there a max length to this? what if someone tries to pass a very long calldata, wouldn't we end up with a comically long ids?

Reply:
Nope, no max length for this one. The longer the input, the longer the output

Reply:
but, will this be a problem?

Reply:
maybe not directly, but this may inflate the file size, and make the file less human-readable.

Reply:
I'll take a look into hashing functions with a fixed-size output.